compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -142,14 +140,14 @@ impl<'a> Resolver<'a> {
         };
 
         // Allocate and return a new module with the information we found
-        let kind = ModuleKind::Def(DefKind::Mod, def_id, name);
-        let module = self.arenas.alloc_module(ModuleData::new(
+        let module = self.arenas.new_module(
             parent,
-            kind,
-            def_id,
+            ModuleKind::Def(DefKind::Mod, def_id, name),
             self.cstore().module_expansion_untracked(def_id, &self.session),
             self.cstore().get_span_untracked(def_id, &self.session),
-        ));
+            // FIXME: Account for `#[no_implicit_prelude]` attributes.
+            parent.map_or(false, |module| module.no_implicit_prelude),
```

Comment:
For enums and traits there should be no consequences (we are jumping out of these modules during resolution and don't use their `no_implicit_prelude` flags), for foreign modules there may be some effect on macros 2.0 - identifiers from macros with def-site located in a foreign `no_implicit_prelude` module should now skip prelude during resolution.

However, because the `#[no_implicit_prelude]` attributes are still not applied to foreign modules the inherited value is always `false`, so no change in behavior right now.